lib/client.js

```diff
@@ -656,29 +586,24 @@ function Client(options) {
                  * @event disconnect
                  */
                 self.emit('disconnect');
-
-                self.clearAutoreconnectTimeout();
-
-                if (!self.autoreconnect) {
-                    return;
-                }
-                if (!self.tokens['App-Secret']) {
-                    self.emit('error', new errors.MissingAppSecretDuringAutoreconnect());
-                    return;
-                }
-                self.autoreconnectTimeout = setTimeout(function () {
-                    self._attemptAutoreconnect();
-                }, self.autoreconnectInterval);
             });
             self.socket.on('reconnect', function (num) {
                 debug('event reconnect', self.endpointId, num);
-                /**
-                 * Reconnected with respoke.
-                 * @event reconnect
-                 * @property {number} num
-                 */
-                self.emit('reconnect', num);
-                self.clearAutoreconnectTimeout();
+                self.wsCall('post', '/connections', dataBody)
```

Comment:
extract a function that both `on('reconnect'...` and `once('connect'...` call to reduce duplication